lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -590,4 +590,58 @@ def thermal_zone_prm_lab_delta_t(thermal_zone)
   def model_get_fan_power_breakdown
     return true
   end
+
+  def model_set_spm(model, thermalZones, coil)
+    # search for the highest zone setpoint temperature
+    max_heat_setpoint = 0.0
+    coil_name = coil.name.get.to_s
+    thermalZones.each do |zone|
+      tstat = zone.thermostatSetpointDualSetpoint
+      if tstat.is_initialized
+        tstat = tstat.get
+        setpoint_sch = tstat.heatingSetpointTemperatureSchedule
+        if setpoint_sch.is_initialized
+          setpoint_sch = setpoint_sch.get
+          if setpoint_sch.to_ScheduleRuleset.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleRuleset.get
+            setpoint_c = schedule_ruleset_annual_min_max_value(setpoint_sch)['max']
+          elsif setpoint_sch.to_ScheduleConstant.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleConstant.get
+            setpoint_c = schedule_constant_annual_min_max_value(setpoint_sch)['max']
+          elsif setpoint_sch.to_ScheduleCompact.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleCompact.get
+            setpoint_c = schedule_compact_annual_min_max_value(setpoint_sch)['max']
+          end
+          if setpoint_c > max_heat_setpoint
+            max_heat_setpoint = setpoint_c
+          end
+        end
+      end
+    end
+
+    max_heat_setpoint_f = OpenStudio.convert(max_heat_setpoint, 'C', 'F').get
+    preheat_setpoint_f = max_heat_setpoint_f - 20
+    preheat_setpoint_c = OpenStudio.convert(preheat_setpoint_f, 'F', 'C').get
+    # in this situation, we hard set the temperature to be 52 F or 11.1 C
+    # (ASHRAE 90.1 Room heating stepoint temperature is 72 F)
+    preheat_setpoint_c = 11.1 if preheat_setpoint_c == 0.0
+
+
+    # create a new constant schedule and this method will add schedule limit type
+    preheat_coil_sch = model_add_constant_schedule_ruleset(model,
+                                                           preheat_setpoint_c,
+                                                           name = "#{coil_name} Setpoint Temp - #{preheat_setpoint_f.round}F")
+    preheat_coil_manager = OpenStudio::Model::SetpointManagerScheduled.new(model, preheat_coil_sch)
+    preheat_coil_manager.setName("#{coil_name} Preheat Coil Setpoint Manager")
+
+    if coil.to_CoilHeatingWater.is_initialized
+      preheat_coil_manager.addToNode(coil.airOutletModelObject.get.to_Node.get)
+    elsif coil.to_CoilHeatingElectric.is_initialized
+      preheat_coil_manager.addToNode(coil.outletModelObject.get.to_Node.get)
+    elsif coil.to_CoilHeatingGas.is_initialized
+      preheat_coil_manager.addToNode(coil.airOutletModelObject.get.to_Node.get)
```

Comment:
I understand that you have this to align with the methods that create systems in `Prototype.Model.rb`. I think we should add a warning message here saying that preheat coils in the baseline should only be hydronic or electric and perhaps add a reference to the PRM-RM. The warning should technically never appear but if for some reason it would, it'll be nice to know that there's an issue with the baseline system creation via the log file.

Reply:
Added OpenStudio::Warn